consensus/src/txn_notifier.rs

```diff
@@ -78,9 +78,10 @@ impl TxnNotifier for MempoolNotifier {
         let user_txn_status = &compute_status[1..txns.len() + 1];
         for (txn, status) in txns.iter().zip_eq(user_txn_status) {
             if let TransactionStatus::Discard(_) = status {
-                rejected_txns.push(TransactionSummary {
+                rejected_txns.push(RejectedTransactionSummary {
                     sender: txn.sender(),
                     sequence_number: txn.sequence_number(),
+                    hash: txn.clone().committed_hash(),
```

Comment:
this hash seems not cached?

Reply:
We need to compare the transaction's hash on rejection notification to the cached value in hash_index. So we'll need to recompute this on rejection.